src/bp/core/services/hook/hook-service.ts

```diff
@@ -21,15 +21,17 @@ const DEBOUNCE_DELAY = ms('2s')
 
 interface HookOptions {
   timeout: number
+  throwOnError: boolean
 }
 
 const debugInstances: { [hookType: string]: IDebugInstance } = {}
+const defaultHookOptions = Object.freeze({ timeout: 1000, throwOnError: false })
```

Comment:
Just a bit curious of why freezing the object is necessary ?

Reply:
It is not _necessary_ per say, but it allows the default options to be readonly.